### PR TITLE
Implement ProxyCustomizer to make real address:port pairs accessible for proxied requests

### LINKED ISSUE #4842: Make PROXY protocol local address optional
**Jetty version**
9.4.x

**Description**
In a setup where client C talks to a server S via a proxy P, and the proxy uses the PROXY protocol towards the server, it may be desirable that the local address is not overridden.

Given:

`C_ip:port -> P_server_ip:port -> P_client_ip:port -> S_ip:port`

In the typical case with the PROXY protocol, the application on the server sees `remote=C_ip:port` and `local=P_server_ip:port`.

However, there are cases where this is desired instead: `remote=C_ip:port` and `local=S_ip:port`.

`ProxyConnectionFactory` could perhaps be enhanced to optionally discard the PROXY protocol data that ends up, on the server, as the `local` address.

### PR DESCRIPTION
**Pull request by lorban**
Fixes https://github.com/eclipse/jetty.project/issues/4842

### CODE REVIEW

File: jetty-server/src/main/java/org/eclipse/jetty/server/ProxyCustomizer.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
```
Comment (olamy) [inline jetty-server/src/main/java/org/eclipse/jetty/server/ProxyCustomizer.java:17]: license header is not correct for 9.4.x :)

File: jetty-server/src/test/java/org/eclipse/jetty/server/ProxyCustomizerTest.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
+package org.eclipse.jetty.server;
```
Comment (olamy) [inline jetty-server/src/test/java/org/eclipse/jetty/server/ProxyCustomizerTest.java:19]: missing license header so the build failure :)

Comment (lorban) [reply jetty-server/src/test/java/org/eclipse/jetty/server/ProxyCustomizerTest.java:19]: Indeed. I should pay more attention to that.

File: jetty-server/src/main/java/org/eclipse/jetty/server/ProxyConnectionFactory.java [hunk 3]
```diff
@@ -764,6 +764,11 @@ public ProxyEndPoint(EndPoint endp, InetSocketAddress remote, InetSocketAddress
             _local = local;
         }
 
+        public EndPoint unwrap()
```
Comment (gregw) [inline jetty-server/src/main/java/org/eclipse/jetty/server/ProxyConnectionFactory.java:767]: What do we call the equivalent method in `DecryptedEndPoint`?
We should call it the same... maybe have a `WrappedEndPoint` common interface/class?

**Review by gregw — APPROVED** [review]
LGTM other than a naming quibble

File: jetty-server/src/main/java/org/eclipse/jetty/server/ProxyConnectionFactory.java (hunk 3 above)
Comment (lorban) [reply jetty-server/src/main/java/org/eclipse/jetty/server/ProxyConnectionFactory.java:767]: There is no equivalent method in `DecryptedEndPoint` as `SecureRequestCustomizer` is only interested in the `SslConnection` in that case.

**Pull request merged**